scipy/stats/_discrete_distns.py

```diff
@@ -867,14 +868,13 @@ def _rvs1(M, n, r, size, random_state):
         return _rvs1(M, n, r, size=size, random_state=random_state)
 
     def _logpmf(self, k, M, n, r):
-        cond = ((r == 0) & (k == 0))
-        result = _lazywhere(~cond, (k, M, n, r),
-                            lambda k, M, n, r:
-                                (-betaln(k+1, r) + betaln(k+r, 1) -
-                                 betaln(n-k+1, M-r-n+1) + betaln(M-r-k+1, 1) +
-                                 betaln(n+1, M-n+1) - betaln(M+1, 1)),
-                            fillvalue=0.0)
-        return result
+        return xpx.apply_where(
```

Comment:
Maybe I'm confusing myself, but did this change the order? I thought the intent was to simplify
```python3
cond = ((r == 0) & (k == 0))
~cond
```
to 
```python3
(r != 0) | (k != 0)
```
but I see that the order of the arguments also changed.